File: python/longCommon.py
#Longest common prefix

class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        strs.sort()
        prefix = strs[0]

        for i in range(len(prefix)):
            if not len(prefix):
                return ""

            else:
                for word in strs:
                    if word[i] != prefix[i]:
                        return prefix[:i]
                    
        
        return prefix

# Working solution

File: cpp/longCommon.cpp
//Longest common prefix

#include <iostream>
#include <vector>
#include <string.h>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix;
        prefix = strs[0];

        int prefixLength = size(prefix);

        for (int i = 0; i < prefixLength; i++)
        {
            if (size(prefix) == 0)
            {
                return "";
            }

            else
            {
                for (int j = 0; j < size(strs); j ++)
                {
                    if (strs[j][i] != prefix[i])
                    {
                        return prefix.substr(0,i);
                    }
                }
            }
        }

        return prefix;
    }
};

//Working solution